freqtrade/rpc/api_server.py

```diff
@@ -0,0 +1,339 @@
+import logging
+import threading
+from datetime import date, datetime
+from ipaddress import IPv4Address
+from typing import Dict
+
+from arrow import Arrow
+from flask import Flask, jsonify, request
+from flask.json import JSONEncoder
+from werkzeug.serving import make_server
+
+from freqtrade.__init__ import __version__
+from freqtrade.rpc.rpc import RPC, RPCException
+
+logger = logging.getLogger(__name__)
+
+BASE_URI = "/api/v1"
+
+
+class ArrowJSONEncoder(JSONEncoder):
+    def default(self, obj):
+        try:
+            if isinstance(obj, Arrow):
+                return obj.for_json()
+            elif isinstance(obj, date):
+                return obj.strftime("%Y-%m-%d")
+            elif isinstance(obj, datetime):
+                return obj.strftime("%Y-%m-%d %H:%M:%S")
+            iterable = iter(obj)
+        except TypeError:
+            pass
+        else:
+            return list(iterable)
+        return JSONEncoder.default(self, obj)
+
+
+class ApiServer(RPC):
+    """
+    This class runs api server and provides rpc.rpc functionality to it
+
+    This class starts a none blocking thread the api server runs within
+    """
+
+    def safe_rpc(func):
```

Comment:
the function is called `safe` which is expected to do some security check. but it is merely catching exceptions. probably another name?